src/test/java/com/exadel/movie_new/tests/MovieServiceTest.java

```diff
@@ -0,0 +1,158 @@
+        when(movieDaoImplMock.read()).thenReturn(list);
+        assertThrows(MovieAlreadyExists.class, () -> movieService.addMovie(newMovie));
+    }
+
+    @Test
+    @DisplayName("Should fetch all movies")
+    void shouldGetAllMovies() {
+        List<Movie> list = new ArrayList<>();
+        list.add(newMovie);
+        list.add(newMovie2);
+        when(movieDaoImplMock.read()).thenReturn(list);
+        List<Movie> movies = movieService.getAllMovies();
+        assertNotNull(movies);
+        assertFalse(movies.isEmpty());
+        assertEquals(2, movies.size());
+    }
+
+    @Test
+    @DisplayName("Should return an empty list")
+    void shouldGetEmptyList() {
+        List<Movie> list = new ArrayList<>();
+        when(movieDaoImplMock.read()).thenReturn(list);
+        List<Movie> movies = movieService.getAllMovies();
+        assertNotNull(movies);
+        assertTrue(movies.isEmpty());
+        assertEquals(0,movies.size());
+    }
+
+    @Test
+    @DisplayName("Should fetch movie by Id")
+    void shouldGetMovieById() {
+        List<Movie> list = new ArrayList<>();
+        list.add(newMovie);
+
+        when(movieDaoImplMock.read()).thenReturn(list);
+        Movie existingMovie = movieService.getMovie(newMovie.getId());
+        assertNotNull(existingMovie);
+        assertNotEquals(null, existingMovie.getId());
+        assertEquals(newMovie.getId(), existingMovie.getId());
+    }
+
+    @Test
+    @DisplayName("Should throw MovieNotFoundException while getMovie")
+    void shouldGetMovieByIdException() {
+        List<Movie> list = new ArrayList<>();
+        list.add(newMovie);
+        when(movieDaoImplMock.read()).thenReturn(list);
+        assertThrows(MovieNotFoundException.class, () -> movieService.getMovie("200"));
+    }
+
+    @Test
+    @DisplayName("Should update movie")
+    void shouldUpdateMovie() {
+        List<Movie> list = new ArrayList<>();
+        list.add(newMovie);
+        when(movieDaoImplMock.read()).thenReturn(list);
+        doNothing().when(movieDaoImplMock).write(any());
+        newMovie.setTitle("Mad Max");
+        Movie existingMovie = movieService.updateMovie(newMovie);
+        assertNotNull(existingMovie);
+        assertEquals("Mad Max", existingMovie.getTitle());
+    }
+
+    @Test
+    @DisplayName("Should thrown MovieNotFoundException while updateMovie")
+    void shouldUpdateMovieException() {
+        List<Movie> list = new ArrayList<>();
+        when(movieDaoImplMock.read()).thenReturn(list);
+        assertThrows(MovieNotFoundException.class, () -> movieService.updateMovie(newMovie));
+    }
+
+    @Test
+    @DisplayName("Should delete movie")
+    void shouldDeleteMovie() {
+        List<Movie> list = new ArrayList<>();
+        list.add(newMovie);
+        list.add(newMovie2);
+        String movieId = "100";
+        when(movieDaoImplMock.read()).thenReturn(list);
+        doNothing().when(movieDaoImplMock).write(any());
+        movieService.deleteMovie(movieId);

```

Comment:
Testing **delete** you need to verify that updated list will be saved: 
verify(movieService).write(**listWithOnlyOneEntity**). That is main idea of the test.  Because currently you just check that your method **deleteMovie** can remove one record from the list, but we need to be sure that exactly this list will saved